src/seamless-immutable.js

```diff
@@ -471,8 +471,17 @@
     return makeImmutable(obj, mutatingObjectMethods);
   }
 
+  var REACT_ELEMENT_TYPE = typeof Symbol === 'function' && Symbol.for && Symbol.for('react.element');
+  var REACT_ELEMENT_TYPE_FALLBACK = 0xeac7;
+
+  function isReactElement(obj) {
+    return obj.hasOwnProperty &&
+           obj.hasOwnProperty('$$typeof') &&
+           obj.$$typeof === REACT_ELEMENT_TYPE_FALLBACK || obj.$$typeof === REACT_ELEMENT_TYPE;
+  }
```

Comment:
This is looking like the right approach! Would you mind adding some comments to the detection linking to where the magic constants come from?

I tried to avoid special casing React components for a long time, but the overwhelming demand has convinced me otherwise. One of the things this means is that now we need to document which React version(s) are supported, and the test cases will need to back that up.

So it probably won't be as simple as just adding React as a single dev dependency; since React will keep having new releases, there will pretty quickly need to be multiple React versions supported and tested separately!